src/main/kotlin/org/rust/ide/template/postfix/PrintlnPostfixTemplate.kt

```diff
@@ -0,0 +1,140 @@
+            }
+            if (addTrailingSemicolon) {
+                macroExpression.add(psiFactory.createSemicolon())
+            }
+            return macroExpression
+        }
+
+        fun replaceWithMacro(expression: PsiElement, addTrailingComma: Boolean = false) {
+            val newElement = expression.replace(createMacro(expression.text, expression.parent is RsBlock))
+            if (addTrailingComma) {
+                newElement.add(psiFactory.createComma())
+            }
+            editor.caretModel.moveToOffset(newElement.endOffset)
+        }
+    }
+
+    override fun expandForChooseExpression(expression: PsiElement, editor: Editor) {
+        if (expression !is RsExpr) return
+
+        val psiFactory = RsPsiFactory(expression.project)
+        val macroCreator = MacroCreator(editor, psiFactory, macroName, Fmt.fromExpr(expression))
+
+        val parent = expression.parent
+        when (parent) {
+            is RsLetDecl -> {
+                val expressionText = parent.pat?.text ?: return
+                addMacroNextToElement(macroCreator.createMacro(expressionText), parent, psiFactory, editor)
+                if (parent.semicolon == null) {
+                    parent.add(psiFactory.createSemicolon())
+                }
+            }
+            is RsMatchArm -> {
+                val matchBody = parent.ancestorStrict<RsMatchBody>() ?: return
+                if (matchBody.containsUnitArm) {
+                    macroCreator.replaceWithMacro(expression, parent.comma == null)
+                } else {
+                    surroundWithBlockExpression(macroCreator.createMacro(expression.text), expression, psiFactory, editor)
+                }
+            }
+            is RsBinaryExpr -> addMacroNextToElement(macroCreator.createMacro(expression.text), parent, psiFactory, editor)
+            else -> macroCreator.replaceWithMacro(expression)
+        }
+    }
+
+    private fun addMacroNextToElement(macro: RsExpr, element: PsiElement, psiFactory: RsPsiFactory, editor: Editor) {
+        val childToContainer = element.closestContainerAncestor ?: return
+        if (childToContainer.second is RsMatchArm) {
+            surroundWithBlockExpression(macro, childToContainer.first, psiFactory, editor)
+        } else {
+            val newElement = childToContainer.second.addAfter(macro, childToContainer.first.getNextNonCommentSibling()?.prevSibling)
+            editor.caretModel.moveToOffset(newElement.endOffset)
+            newElement.add(psiFactory.createNewline())
+        }
+    }
+
+    private fun surroundWithBlockExpression(macro: RsExpr, originalExpression: PsiElement, psiFactory: RsPsiFactory, editor: Editor) {
+        val newBlock = psiFactory.createBlockExpr("${macro.text}\n${originalExpression.text}")
+        val newElement = originalExpression.replace(newBlock) as RsBlockExpr
+        editor.caretModel.moveToOffset(newElement.block.children[0].endOffset)
```

Comment:
Looks like you should reformat the new block, for example, for such case:
```
let x = match 42 {
    0..10 => 5 + 7.println * 2
};
```
I think you can just put
```Kotlin
CodeStyleManager.getInstance(newElement.project).reformat(newElement.block.ancestorStrict<RsBlock>()?: newElement.block)
```
at the end of this function.